db/migrations/20200615231106_create_median_storage_tables.sql

```diff
@@ -0,0 +1,116 @@
+-- +goose Up
+CREATE TABLE maker.median_val
+(
+    id        SERIAL PRIMARY KEY,
+    diff_id   BIGINT  NOT NULL REFERENCES storage_diff (id) ON DELETE CASCADE,
+    header_id INTEGER NOT NULL REFERENCES headers (id) ON DELETE CASCADE,
+    address_id INTEGER NOT NULL REFERENCES addresses (id) ON DELETE CASCADE,
+    val       NUMERIC NOT NULL,
+    UNIQUE (diff_id, header_id, address_id, val)
+);
+
+CREATE INDEX median_val_diff_id_index
+    ON maker.median_val (diff_id);
+CREATE INDEX median_val_header_id_index
+    ON maker.median_val (header_id);
+CREATE INDEX median_val_address_id_index
+    ON maker.median_val (address_id);
+
+CREATE TABLE maker.median_bar
+(
+    id        SERIAL PRIMARY KEY,
+    diff_id   BIGINT  NOT NULL REFERENCES storage_diff (id) ON DELETE CASCADE,
+    header_id INTEGER NOT NULL REFERENCES headers (id) ON DELETE CASCADE,
+    address_id INTEGER NOT NULL REFERENCES addresses (id) ON DELETE CASCADE,
+    bar       NUMERIC NOT NULL,
+    UNIQUE (diff_id, header_id, address_id, bar)
+);
+
+CREATE INDEX median_bar_diff_id_index
+    ON maker.median_bar (diff_id);
+CREATE INDEX median_bar_header_id_index
+    ON maker.median_bar (header_id);
+CREATE INDEX median_bar_address_id_index
+    ON maker.median_bar (address_id);
+
+CREATE TABLE maker.median_age
+(
+    id        SERIAL PRIMARY KEY,
+    diff_id   BIGINT  NOT NULL REFERENCES storage_diff (id) ON DELETE CASCADE,
+    header_id INTEGER NOT NULL REFERENCES headers (id) ON DELETE CASCADE,
+    address_id INTEGER NOT NULL REFERENCES addresses (id) ON DELETE CASCADE,
+    age       NUMERIC NOT NULL,
+    UNIQUE (diff_id, header_id, address_id, age)
+);
+
+CREATE INDEX median_age_diff_id_index
+    ON maker.median_age (diff_id);
+CREATE INDEX median_age_header_id_index
+    ON maker.median_age (header_id);
+CREATE INDEX median_age_address_id_index
+    ON maker.median_age (address_id);
+
+CREATE TABLE maker.median_orcl
+(
+    id         SERIAL PRIMARY KEY,
+    diff_id    BIGINT  NOT NULL REFERENCES storage_diff (id) ON DELETE CASCADE,
+    header_id  INTEGER NOT NULL REFERENCES headers (id) ON DELETE CASCADE,
+    address_id INTEGER NOT NULL REFERENCES addresses (id) ON DELETE CASCADE,
+    a          INTEGER NOT NULL REFERENCES addresses (id) ON DELETE CASCADE,
+    orcl        INTEGER NOT NULL,
+    UNIQUE (diff_id, header_id, address_id, a, orcl)
```

Comment:
This would be different from what we did previously, _but_ I've been thinking maybe we should omit the `diff_id` from the uniqueness constraint on these tables. Unique on `header_id` + `address_id` + `a` + `orcl` would still prevent those duplicates, but _also_ catch cases where (for whatever reason) we're getting multiple diffs for the same state change